Prozentrechner modul dev

flakeykid opened the pull request.

Commit:
ProzentrechnungKlasse Prozent dazu und weg rechnen, Netto und Bruttopreis

Commit:
Merge remote-tracking branch 'origin/develop' into prozentrechner_modul_DEV

Commit:
Unittests Prozentrechner

Commit:
Unittests + Fixes Taschenrechnerklasse

Commit:
Prozentrechnung Unittests Nettopreis, Bruttopreis und Prozentsatz

Commit:
Prozentsatz fix + Unittests Prozentsatz

FabiusGruener commented on Classes/Prozentrechnung.cs:
```diff
@@ -0,0 +1,75 @@
+    {
+        public static double ProzentDazu (double psatz, double grundwert)
+        {
+            double ergebnis = 1;
```
ergebnis = grundwert, wenn nichts dazu kommt

FabiusGruener commented on Classes/Prozentrechnung.cs:
```diff
@@ -0,0 +1,75 @@
+
+        public static double ProzentWeg(double prozentsatz, double grundwert)
+        {
+            double ergebnis = 1;
```
ergebnis = grundwert, wenn nichts abgezogen wird

FabiusGruener commented on Classes/Prozentrechnung.cs:
```diff
@@ -0,0 +1,75 @@
+            {
+                return 1;
+            }
+            else
```
die Abfrage ist unnötig, da der Nettopreis auch 0 ist, wenn der Bruttopreis 0 ist.

FabiusGruener commented on Classes/Prozentrechnung.cs:
```diff
@@ -0,0 +1,75 @@
+            }
+            else
+            {
+                double ergebnis = bruttopreis * 0.81;
```
Bruttopreis = 119% vom Nettopreis, also:
ergebnis = bruttopreis * 100 / 119

FabiusGruener commented on Classes/Prozentrechnung.cs:
```diff
@@ -0,0 +1,75 @@
+            {
+                return 1;
+            }
+            else
```
If-Abfrage unnötig, bzw. falsch: wenn Nettopreis 0 ist, ist der Bruttopreis auch 0

FabiusGruener commented on Classes/Prozentrechnung.cs:
```diff
@@ -0,0 +1,75 @@
+        }
+        public static double Prozentsatz(double prozentwert, double grundwert)
+        {
+            double ergebnis = 1;
```
Hier würde ich sowas wie -1 ausgeben, da es sonst nicht vorkommt und dann bei Aufruf der Funktion entsprechend behandeln
edit: also wenn grundwert gleich 0 ist

FabiusGruener reviewed (commented):
hab paar Kommentare geschrieben

nasskalt commented on TaschenrechnerUnitTests/Prozentrechnung/ProzentDazu.cs:
```diff
@@ -0,0 +1,26 @@
+        }
+
+        [Test]
+        public void Input_0()
```
bessere funktionsbennung in den test klassen wäre schön

Commit:
Änderungen in der Klasse laut Review übernommen
Funktionsbenennung in den Klassen angepasst (bsp.: "Input_Prozentsatz_0_Grundwert_0")

Commit:
some changes for optimisation

Commit:
im optimalfall
